std/internal/math/mathnoasm.d

```diff
@@ -0,0 +1,154 @@
+// Written in the D programming language.
+
+/* Contains elementary mathematical functions, and low-level
+ * floating-point operations for processors with no asm support.
+ *
+ * All of these functions are subject to change, and are intended
+ * for internal use only.
+ *
+ * Copyright: Copyright The D Language Foundation 2000 - 2011.
+ * License:   $(HTTP www.boost.org/LICENSE_1_0.txt, Boost License 1.0).
+ * Authors:   $(HTTP digitalmars.com, Walter Bright), Don Clugston,
+ *            Conversion of CEPHES math library to D by Iain Buclaw and David Nadlinger
+ * Source: $(PHOBOSSRC std/internal/math/mathnoasm.d)
+ */
+module std.internal.math.mathnoasm;
+
+static import core.math;
+import std.math : floatTraits, RealFormat, PI, PI_4,
+       isNaN, isInfinity, signbit, floor, poly;
+
+/////////////////////////////////////////////////////////////////////////////
+
+T tan(T)(T x) @safe pure nothrow @nogc
+{
+    // Coefficients for tan(x) and PI/4 split into three parts.
+    enum realFormat = floatTraits!T.realFormat;
+    static if (realFormat == RealFormat.ieeeQuadruple)
```

Comment:
Since we're in the process of making this nicer, can we move these out onto a (private) mixin template?

Reply:
Like this?

Reply:
Yep.